### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

**Comment by ajschmidt8** [comment]
Is this a duplicate of #680?

File: .github/labeler.yml [hunk 1]
```diff
@@ -4,22 +4,10 @@
 
 Python:
   - 'python/**'
-  - 'notebooks/**'
-  
-librmm:
-  - 'cpp/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:10]: I believe this should be maintained, @harrism what label should we use here if any?

File: .github/labeler.yml [hunk 2]
```diff
@@ -4,22 +4,10 @@
...
-  
-Java:
-  - 'java/**'
-  
-Ops:
-  - '.github/**'
-  - 'ci/**'
+
+conda:
   - 'conda/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:13]: I'd add gpuCI to match cuDF and other repos
```suggestion
  - 'conda/**'

gpuCI:
  - 'ci/**'
```

**Review by mike-wendt — CHANGES_REQUESTED** [review]

**Comment by mike-wendt** [comment]
@jrhemstad @harrism I think there may be some missing labels here but can you provide feedback so we can get this in for 0.18?

File: .github/labeler.yml (hunk 1 above)
Comment (kkraus14) [reply .github/labeler.yml:10]: We should probably create a `cpp` label for this

**Comment by harrism** [comment]
> @jrhemstad @harrism I think there may be some missing labels here but can you provide feedback so we can get this in for 0.18?

There is not enough information in the description and the changes in this PR for me to understand exactly what it does. Examples would help.

**Comment by ajschmidt8** [comment]
> > @jrhemstad @harrism I think there may be some missing labels here but can you provide feedback so we can get this in for 0.18?
> 
> There is not enough information in the description and the changes in this PR for me to understand exactly what it does. Examples would help.

@harrism, this PR adds some automation to label PRs when they're opened. As an example, the configuration below will automatically label any PRs that edit files in the `python` directory with the `Python` label.

```yaml
Python:
  - 'python/**'
```

File: .github/labeler.yml [hunk 3]
```diff
@@ -12,5 +12,8 @@ CMake:
 conda:
   - 'conda/**'
 
+cpp:
+  - 'cpp/**'
```
Comment (kkraus14) [inline .github/labeler.yml:16]: We don't currently have a `cpp` folder in the rmm repo. This should be `include/` unless we want to reorganize the code.

Comment (kkraus14) [reply .github/labeler.yml:16]: or rather should include `include/`, `tests/` and `doxygen/`

Comment (harrism) [reply .github/labeler.yml]: Agree.

**Review by harrism — APPROVED** [review]

**Review by mike-wendt — APPROVED** [review]

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
note we still need to make a `cpp` label

**Comment by mike-wendt** [comment]
> note we still need to make a `cpp` label

@kkraus14 it will get auto-generated just without a color; just like the `Ops` label (we'll remove that after the config is updated otherwise it will be recreated)

**Comment by harrism** [comment]
Is this ready to merge? @jolorunyomi @mike-wendt

**Comment by ajschmidt8** [comment]
removing the labels from this PR to prevent it from appearing in the changelog since it was re-merged correctly (squash merged) in #691.

**Pull request merged**